Gordon360/Controllers/MembershipsController.cs

```diff
@@ -218,7 +233,7 @@ public async Task<ActionResult<MembershipView>> SetGroupAdminAsync(int membershi
         /// <param name="isPrivate">The new value of Privacy for the membership</param>
         /// <remarks>Calls the server to make a call and update the database with the changed information</remarks>
         /// <returns>The updated membership as a MembershipView object</returns>
-        [HttpPut]
+        [HttpPatch]
```

Comment:
Why was this changed? This creates a breaking change for the UI, and it seems like it was included in an unrelated pull request?

Reply:
I believe this was a change to update the protocol after I added `patch` (since this is a purely updating route). I believe that this change is reflected in the UI PR.

Reply:
Technically, this should be a PUT route, not a PATCH route, since the resource we're acting on is `/memberships/{membershipID}/privacy`, that is the resource we're PUTting is the privacy field itself, so we aren't updating one specific part of the resource buet rather replacing the whole resource with the given value.

COnversely, if the route were simply `/memberships/{membershipID}`, but we were selectively updating some parts of the membership resource based on the parameters, that would be a proper PATCH method.

More importantly, the better way to handle breaking changes to the API is to mark the old route as obsolete (but leave it unchanged), and create the new route. Then, once all clients (in our case, the develop UI) has updated, we can remove the obsolete. That way, deploying the API will not break the UI. As it is currently, we would have to coordinate the upgrade of the API and UI so that they both update in tandem. That is not infeasible in our environment, although it can be a hassle because both have to wait on the other to be ready before merging. In larger-scale projects or APIs with multiple/third-party consumers, that can be the cause of a major outage.